### ISSUE TITLE
[Suggestion]: Some config and other suggestions!

### ISSUE THREAD

**Issue by Ultimushadow** [Fuzss/illagerinvasion#19]
### Suggestion (Required)

Hey there, I love the mod! I am currently working on a pack and noticed the config doesn't touch a lot of the features in the mod, so I just wanted to throw a handful of ideas out there, some better than others, and mix in some potential changes/additions to the mod

- Move enchantment tag to config: While tags and datapacks are easy to understand and use, it's more involved than opening up the config file. Obviously the trade off is requiring a restart rather than changing things during run time, but it is more user friendly!
- Trim effect toggle per piece: Currently the effects are all on or all off, which is less configurable
- Add secondary boot effect option: Trampling crops is a _very_ commonly addressed area in modding and thus this makes the effect essentially useless if any of those mods are installed. The other trims don't have this issue as even if another mod adds the same effect, they would simply stack. I suggest adding a small secondary effect (like an additional block of building reach) and the boots can just have both by default. Trample protection is not on the same level of power or usefulness as the other anyways, so this seems like a good idea just to balance things out a bit. Config option here would be disabling one effect or the other, as some mods tie trample to feather falling so this avoids feature overlap if the user wishes it.
- Number of Imbuements: I think allowing the Imbuing table to go more than once beyond the max level would be invaluable for pack makers, as well as possibly specifying the item that needs to be used in order to be used for the imbuement. The extended number of imbuements I don't think would be too much of an issue from code, though specifying an item would be difficult if the max level is uncapped for the user to specify. In the spirit of more freedom I would just require the amount of gems in either powers of 2 or Fibonacci increments, so later upgrades cost more until you hit 64. 
- Imbuement table enchantment power requirement: Just an implementation of the Enchanting Table power requirements of bookshelves, personally I think a block as powerful as this deserves its own special spot! By configuring how much is needed, the user could require players to use boosted bookshelves from other mods!
- Candle ores: If I'm reading this correctly, you do check block tags here but only for the vanilla ores. The ability to add our own ores and specify the blocks that correspond would be great for compatibility!
- Purple fire overlay: When standing in fire, one normally gets a fire overlay. Mods like Soul Fire'd even change this to blue for Soul Fire, so I figure it would be nice to keep the consistency
- Another use for magical fire: I love the purple flames but they're basically just decoration and as far as I know not obtainable in survival? At least as far as EMI is telling me. Do you plan on expanding this? Perhaps an enchantment for the Infused Hatchet which lets it deal bonus fire/magic damage?
- Hatchet improvements: Speaking of the hatchet it's unfortunately not very friendly to use and does not make up for it with any benefits as far as I can tell. First, I'm not sure it's even obtainable? It's not in the Marauder's loot table, it's not a smithing table recipe with a netherite axe/platinum ingot, there are no loot tables with it, I checked the discord and there wasn't even a mention of the hatchet there, maybe I'm being dumb? Either way, however you get it you should probably make that information a little more prominent lol. Besides that though, the hatchet has potential to be a high damage alternative to the Trident but falls short due to the lack of enchantments, both damaging and utility. It could benefit from some kind of special effect (like the arcane fire idea) and an easier way to get the thing back. Maybe just keep the item in the inventory and make it cooldown based, with an enchantment to lower the cooldown slightly? Also, it would be interesting if it scaled off melee damage, so things like Sharpness and Strength would actually effect the projectile.
- More magic: Overall, most items in the mod are used to make a single thing, and there isn't a ton of cohesion even within the mod itself. Primal essence is used to make one item, illusionary and unusual dust are used to make one item, which itself has a single use, platinum chunks are used only to make the sheet, horn and candle items are only found in structures rather than interfacing with the other items, the axe and fire charge are unobtainable, the fire only deals damage, the imbuement table has a single purpose. At least the sheets can be used as a trim with multiple buffs, but that is also basically one use. I understand the mod is not far into development so I am unsure of your plans but I can see a lot of these working together for some very interesting gameplay
       - Illusionary dust being used to craft a player duplicate to confuse mobs
       - Unusual dust being used in a recipe to craft Magical Fire Charges
       - Primal Essence and Hallowed Gems used in conjunction with the dusts and Platinum for crafting tools/weapons with weaker versions of the attacks the mobs use on the player
       - Imbuement Table being used to empower the above suggested items to be a little stronger
       - Platinum being used in to craft blinders for a ravager so the player can ride them
       - Platinum used for reinforcing shields to tank more axe hits
       - Magical Fire Charges used for crafting Arcane Explosives, special fireworks that can be loaded into a dispenser or crossbow and give the targets a couple seconds of being on Magical Fire
       - Villagers counterparts to some of the Illagers, like a Mystic who will summon Allays in exchange for emeralds and Amethyst, or a Sage who uses water inspired healing abilities
       
Like I said, I got the idea you consider this mod very WIP and I know you were basing it off an older mod so I don't know how much of this is planned or conflicts with that - just putting some thoughts out there in case you find any of them worth considering!

**Comment by Fuzss** [comment]
Thank you so much for all those suggestions and the effort you have put into them!

It will be a while before I come back to working on this mod to be able to address some of them, but just wanted to throw in a quick thank you for taking the time to come up with so much!

**Comment by Ultimushadow** [comment]
Of course, when you have the time and desire to check them out feel free to tag me if some of the ideas need tinkering or playtesting! I appreciate the warning about the timeline a ton 🙏🏻 good luck on your current projects!
